### PR TITLE
[MSGINA] Initialize the GINA context

### PR DESCRIPTION
**Pull request by Fraizeraust**
A regression which prevented the animated bar from loading was introduced by #2353. This patch fixes it.

### CODE REVIEW

**Comment by HBelusca** [comment]
Suggestion:
1. Extend `DlgData_Create()` definition to:
```c
static PDLG_DATA
DlgData_Create(HWND hwndDlg, PGINA_CONTEXT pgContext)
```
and assign `pDlgData->pgContext` to `pgContext` within.
2. Fix all the `DlgData_Create()` calls suitably.

If such a thing had been done in the original PR that bug would not have appeared ^^

File: dll/win32/msgina/gui.c [hunk 1]
```diff
@@ -42,14 +42,23 @@ typedef struct _DLG_DATA
...
+            return NULL;
+        }
+
+        pDlgData->pgContext = pgContext;
         SetWindowLongPtrW(hwndDlg, GWLP_USERDATA, (LONG_PTR)pDlgData);
+        return pDlgData;
     }
-    return pDlgData;
+
+    return NULL;
```
Comment (binarymaster) [inline dll/win32/msgina/gui.c]: I think it's unnecessary change.
